runtime/gc_structs/ConstantPoolObjectSlotIterator.cpp

```diff
@@ -75,19 +86,26 @@ GC_ConstantPoolObjectSlotIterator::nextSlot()
 				result = &(((J9RAMMethodHandleRef *) slotPtr)->methodHandle);
 				break;
 			case J9CPTYPE_CONSTANT_DYNAMIC:
-				if (NULL != ((J9RAMConstantDynamicRef *) slotPtr)->value) {
-					result = &(((J9RAMConstantDynamicRef *) slotPtr)->value);
-				} else {
-					result = &(((J9RAMConstantDynamicRef *) slotPtr)->exception);
-				}
+				result = scanCondySlot(slotPtr, &nextSlot);
```

Comment:
technically we can create another class to iterate objects from single condy slot and move this function there if you like (condySlotObhjecrtIterator?)... shift to next slot when iterator returns NULL. I am neutral about this so it is up to you